core/templates/pages/collection-editor-page/navbar/collection-editor-navbar.directive.spec.ts

```diff
@@ -0,0 +1,549 @@
+      }],
+      playthrough_dict: {
+        next_exploration_id: 'expId',
+        completed_exploration_ids: ['expId2']
+      }
+    };
+
+    sampleCollectionBackendObject2 = {
+      id: 'collection_id',
+      title: null,
+      objective: 'an objective',
+      language_code: 'en',
+      tags: [],
+      category: 'a category',
+      version: 1,
+      schema_version: 1,
+      nodes: [{
+        exploration_id: 'exp_id0',
+        exploration_summary: learnerExplorationSummaryBackendDict
+      },
+      {
+        exploration_id: 'exp_id1',
+        exploration_summary: learnerExplorationSummaryBackendDict
+      }],
+      playthrough_dict: {
+        next_exploration_id: 'expId',
+        completed_exploration_ids: ['expId2']
+      }
+    };
+    sampleCollection = Collection.create(
+      sampleCollectionBackendObject1);
+    sampleCollectionRights = CollectionRights.createEmptyCollectionRights();
+
+    ctrl = $injector.instantiate(directive.controller, {
+      $rootScope: $scope,
+      $scope: $scope
+    });
+  }));
+
+  beforeEach(() => {
+    testSubscriptions = new Subscription();
+    testSubscriptions.add(
+      collectionEditorStateService.onCollectionInitialized.subscribe(
+        collectionInitializedSpy));
+  });
+
+  afterEach(() => {
+    testSubscriptions.unsubscribe();
+    ctrl.$onDestroy();
+  });
+
+  it('should set properties when initialized', function() {
+    spyOn(urlService, 'getCollectionIdFromEditorUrl')
+      .and.returnValue('collectionId');
+    spyOn(collectionEditorStateService, 'getCollection')
+      .and.returnValue(sampleCollection);
+    spyOn(collectionEditorStateService, 'getCollectionRights')
+      .and.returnValue(sampleCollectionRights);
+    spyOn(undoRedoService, 'onUndoRedoChangeApplied$').and.returnValue(
+      mockUndoRedoChangeEventEmitter);
+    spyOnProperty(collectionEditorStateService, 'onCollectionInitialized')
+      .and.returnValue(mockCollectionInitializedEventEmitter);
+
+    ctrl.$onInit();
+    $rootScope.$apply();
+
+    expect(ctrl.collectionId).toBe('collectionId');
+    expect(ctrl.collectionRights).toBe(sampleCollectionRights);
+  });
+
+  it('should validate issues when undo redo change ' +
+    'event is emitted', function() {
+    spyOn(urlService, 'getCollectionIdFromEditorUrl')
+      .and.returnValue('collectionId');
+    spyOn(collectionEditorStateService, 'getCollection')
+      .and.returnValue(sampleCollection);
+    spyOn(collectionEditorStateService, 'getCollectionRights')
+      .and.returnValue(sampleCollectionRights);
+    spyOn(undoRedoService, 'onUndoRedoChangeApplied$').and.returnValue(
+      mockUndoRedoChangeEventEmitter);
+    let validationSpy = spyOn(
+      collectionValidationService, 'findValidationIssuesForPublicCollection')
+      .and.callThrough();
+
+    ctrl.$onInit();
+    $rootScope.$apply();
+    mockUndoRedoChangeEventEmitter.emit();
+    $rootScope.$apply();
```

Comment:
Do we need multiple `$rootScope` stmts here?

Reply:
done.